ext/ddtrace_profiling_native_extension/ruby_helpers.c

```diff
@@ -167,3 +167,39 @@ bool ruby_ref_from_id(VALUE obj_id, VALUE *value) {
 
   return true;
 }
+
+// Not part of public headers but is externed from Ruby
+size_t rb_obj_memsize_of(VALUE obj);
+
+// Wrapper around rb_obj_memsize_of to avoid hitting crashing paths.
+size_t ruby_obj_memsize_of(VALUE obj) {
```

Comment:
To be fair, the crashing path is only when an object is invalid -- all of the Ruby types are accounted for in `rb_obj_memsize_of(...)` so I'm not sure it's worth having this wrapper around it -- `rb_bug(...)` is a pretty big thing that the VM only uses when it truly believes there's something wrong at the VM/native level.

Reply:
There's an extra "hidden" `rb_bug` call for the `T_NODE` case which was the one that worried me a bit more. The fact that they use `rb_bug` in that place may be a hint that it should be impossible for us to accidentally track one. But I'm also not 100% sure if that's the case or if they just assumed whoever called `obj_memsize_of` would be doing that check first.

Reply:
You're right -- I had missed that one. I don't know enough about RNode; intuitively if they're calling `rb_bug` it would be weird if we ever got handed one of those objects but I definitely see your concern.

I guess it may worth leaving a bit more context as a comment on why we're doing this, but I'm convinced :)